modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/notifications/NotificationsInfo.js

```diff
@@ -399,16 +395,16 @@ const NotificationsInfo = ({
 						...previousItem.data.notifications.recipients[
 							notificationIndex
 						],
-						...currentRecipient,
+						...recipientDetails,
 					};
 				}
 				else {
 					previousItem.data.notifications.recipients[
 						notificationIndex
-					] = currentRecipient;
+					] = recipientDetails;
 				}
 
-				return previousItem;
+				return {...previousItem};
```

Comment:
@rebsilva this is the latest change. We should never return `previousItem` as is. We should make sure that state updates get a new object.